elegant/clean_timepoint_data.py

```diff
@@ -0,0 +1,147 @@
+import pathlib
+import shutil
+import json
+import datetime
+
+import numpy
+
+from zplib import datafile
+from elegant import load_data
+
+def remove_timepoint_for_position(experiment_root, position, timepoint, dry_run=False):
+    """Removes all memory of a given timepoint for a given position (files, metadata, annotations entries)
+
+    Parameters
+        experiment_root - str/pathlib.Path pointing to experiment directory
+        position - str for position to modify
+        timepoint - str with standard format for timepoint acquisition (YYYY-MM-DDtHHMM)
+        dry_run - bool flag that toggles taking action (if False, only specifies when offending files are found)
+
+    Example Usage
+
+        experiment_root = /path/to/experiment
+        position = '000'
+        timepoint = '2018-07-30t1200'
+        dry_run = True  # No deleting; just see what happens
+        remove_timepoint_for_position(experiment_root, position, timepoint, dry_run=dry_run)
+    """
+
+    experiment_root = pathlib.Path(experiment_root)
+
+    files_to_remove = [img_file for img_file in (experiment_root / position).iterdir() if timepoint in str(img_file.name)]
+    if len(files_to_remove) > 0:
+        print(f'Found files for removal for position {position}: {files_to_remove}')
+        if not dry_run:
+            [img_file.unlink() for img_file in files_to_remove]
+
+    md_file = (experiment_root / position /'position_metadata.json')
+    with md_file.open() as md_fp:
+        pos_md = json.load(md_fp)
+    has_bad_timepoint = any([timepoint_info['timepoint'] == timepoint for timepoint_info in pos_md])
+    if has_bad_timepoint:
+        print(f'Found entry for removal in position_metadata for position {position}')
+        if not dry_run:
+            pos_md = [timepoint_data for timepoint_data in pos_md if timepoint_data['timepoints'] != timepoint]
```

Comment:
Also, if you do something identical twice, use the same variable names. Here it's `timepoint_data` but before the loop var is `timepoint_info`. Variable-name consistency is important for maintainability. If a variable has different names in different places, I am going to think it is doing different things.